js/compiler.ts

```diff
@@ -167,6 +186,57 @@ export class DenoCompiler implements ts.LanguageServiceHost {
   // testing
   private _window = window;
 
+  /**
+   * Drain the instantation queue, retrieving the arguments for the module
+   * factory and calling the module's factory.
+   */
+  private _drainInstantateQueue(): void {
+    this._log(
+      "compiler._drainInstantateQueue",
+      this._instantiateQueue.map(metaData => metaData.fileName)
+    );
+    let moduleMetaData: ModuleMetaData | undefined;
+    while ((moduleMetaData = this._instantiateQueue.shift())) {
+      assert(
+        moduleMetaData.factory != null,
+        "Cannot instantiate module without factory."
+      );
+      assert(!moduleMetaData.hasRun, "Module has already been instantiated.");
+      // asserts not tracked by TypeScripts, so using not null operator
+      moduleMetaData.factory!(...this._getFactoryArguments(moduleMetaData));
+      moduleMetaData.hasRun = true;
```

Comment:
Consider renaming `_instantiateQueue` to `_runQueue` 
or
renaming `hasRun` to `instantiated` ?

(There's also the `Compiler.run` method. I don't care which term we use as long as it's consistent.)

Reply:
Let's go with `run` it is a little informal, but it makes it clear it is "in memory and active" to just about anyone. I guess instantiate is a bit jargony.